src/main/java/org/jabref/gui/actions/IntegrityCheckAction.java

```diff
@@ -50,7 +54,36 @@ public void actionPerformed(ActionEvent e) {
                 Globals.prefs.getBibtexKeyPatternPreferences(),
                 Globals.journalAbbreviationLoader
                         .getRepository(Globals.prefs.getJournalAbbreviationPreferences()));
-        List<IntegrityMessage> messages = check.checkBibtexDatabase();
+
+        final JDialog integrityDialog = new JDialog(frame, true);
+        integrityDialog.setUndecorated(true);
+        integrityDialog.setLocationRelativeTo(frame);
+        JProgressBar integrityProgressBar = new JProgressBar();
+        integrityProgressBar.setIndeterminate(true);
+        integrityProgressBar.setStringPainted(true);
+        integrityProgressBar.setString("Checking integrity...");
```

Comment:
Please use a Localized message and add it to the l10n files 
https://github.com/JabRef/jabref/wiki/Code-Howtos#using-localization-correctly